.github/workflows/build-image.yml

```diff
@@ -0,0 +1,84 @@
+name: Build Container Image & Publish
+permissions: read-all
+on:
+  push:
+    # Publish semver tags as releases.
+    tags: [v*.*.*]
+
+env:
+  # Use docker.io for Docker Hub if empty
+  REGISTRY: ghcr.io
+  # github.repository as <account>/<repo>
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+      # This is used to complete the identity challenge
+      # with sigstore/fulcio when running outside of PRs.
+      id-token: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      # Install the cosign tool except on PR
+      # https://github.com/sigstore/cosign-installer
+      - name: Install cosign
+        if: github.event_name != 'pull_request'
```

Comment:
If I remember correctly, it's possible to have a single `if` for the entire flow, that'd be better for this case.
Also, instead of checking `!= 'pull_request'` it'd be better to check for merge to `main`.